lib/wallet/src/Cardano/Wallet/DB/Layer.hs

```diff
@@ -1047,9 +1047,11 @@ selectTxHistory cp ti wid minWithdrawal order whichMeta
             (\coin -> any (>= coin)
                 $ txWithdrawalAmount <$>  withdrawals transaction)
             minWithdrawal
+        let decoration = decorateTxIns txHistory transaction
```

Comment:
👌 👍 Types look simpler without `'With` / `'Without` distinction, and it should hopefully be more efficient to only resolve the inputs we actually need (but also don't grasp how/if laziness helps the current approach)

Reply:
>  (but also don't grasp how/if laziness helps the current approach)

Laziness probably helps a lot, but I also don't understand precisely how much. Removing the `'With` / `'Without` distinction was my primary objective.